homeassistant/components/xiaomi_miio/services.yaml

```
@@ -1,365 +1,621 @@
   description: Manipulates a storage register which advertises extra features. The Mi Home app evaluates the value. A feature called "turbo mode" is unlocked in the app on value 1.
   fields:
     entity_id:
       description: Name of the xiaomi miio entity.
-      example: "fan.xiaomi_miio_device"
+      selector:
+        entity:
+          integration: xiaomi_miio
+          domain: fan
     features:
+      name: Features
       description: Integer, known values are 0 (default) and 1 (turbo mode).
-      example: 1
+      required: true
+      selector:
+        number:
+          min: 0
+          max: 1
 
 fan_set_target_humidity:
+  name: Fan set target humidity
   description: Set the target humidity.
   fields:
     entity_id:
       description: Name of the xiaomi miio entity.
-      example: "fan.xiaomi_miio_device"
+      selector:
+        entity:
+          integration: xiaomi_miio
+          domain: fan
     humidity:
-      description: Target humidity. Allowed values are 30, 40, 50, 60, 70 and 80.
-      example: 50
+      name: Humidity
+      description: Target humidity.
+      required: true
+      selector:
+        number:
+          min: 30
+          max: 80
+          step: 10
+          unit_of_measurement: '%'
 
 fan_set_dry_on:
+  name: Fan set dry on
   description: Turn the dry mode on.
   fields:
     entity_id:
       description: Name of the xiaomi miio entity.
-      example: "fan.xiaomi_miio_device"
+      selector:
+        entity:
+          integration: xiaomi_miio
+          domain: fan
 
 fan_set_dry_off:
+  name: Fan set dry off
   description: Turn the dry mode off.
   fields:
     entity_id:
       description: Name of the xiaomi miio entity.
-      example: "fan.xiaomi_miio_device"
+      selector:
+        entity:
+          integration: xiaomi_miio
+          domain: fan
 
 fan_set_motor_speed:
+  name: Fan set motor speed
   description: Set the target motor speed.
   fields:
     entity_id:
       description: Name of the xiaomi miio entity.
-      example: 'fan.xiaomi_miio_device'
+      selector:
+        entity:
+          integration: xiaomi_miio
+          domain: fan
     motor_speed:
-      description: Set RPM of motor speed, between 200 and 2000.
-      example: 1100
+      name: Motor speed
+      description: Set motor speed.
+      required: true
+      selector:
+        number:
+          min: 200
+          max: 2000
+          unit_of_measurement: 'RPM'
 
 light_set_scene:
+  name: Light set scene
   description: Set a fixed scene.
   fields:
     entity_id:
       description: Name of the light entity.
-      example: "light.xiaomi_miio"
+      selector:
+        entity:
+          integration: xiaomi_miio
+          domain: light
     scene:
-      description: Number of the fixed scene, between 1 and 4.
-      example: 1
+      name: Scene
+      description: Number of the fixed scene.
+      required: true
+      selector:
+        number:
+          min: 1
+          max: 4
```

According to the schema it is 6 😬 
```suggestion
          max: 6
```